src/platform/macos.rs

```diff
@@ -1,12 +1,17 @@
 extern crate libc;
+extern crate libloading;
 
 use std::os::unix::ffi::OsStrExt;
 use std::path::Path;
 use std::{io, ffi};
+use self::libloading::os::unix::{Library, Symbol};
 
-extern "stdcall" {
-	fn renamex_np(oldpath: *const libc::c_char, newpath: *const libc::c_char, flags: libc::c_uint) -> libc::c_int;
+unsafe fn get_renamex_np() -> Option<Symbol<unsafe extern fn (oldpath: *const libc::c_char, newpath: *const libc::c_char, flags: libc::c_uint) -> libc::c_int>> {
+	let lib = Library::this();
+	lib.get(b"renamex_np").ok()
```

Comment:
Are we sure we don't want to panic here, `.expect("failed to get renamex_np, cannot proceed.")`?

Reply:
we can proceed, we can `renamex_np` is not available only on osx > `10.12`. if it's not available, we fallback to `exchangedata`

Reply:
nice